src/aws_encryption_sdk/key_providers/kms.py

```diff
@@ -82,22 +82,58 @@ class KMSMasterKeyProvider(MasterKeyProvider):
         Master key providers are deprecated.
         Use :class:`aws_encryption_sdk.keyrings.aws_kms.AwsKmsKeyring` instead.
 
-    >>> import aws_encryption_sdk
-    >>> kms_key_provider = aws_encryption_sdk.KMSMasterKeyProvider(key_ids=[
-    ...     'arn:aws:kms:us-east-1:2222222222222:key/22222222-2222-2222-2222-222222222222',
-    ...     'arn:aws:kms:us-east-1:3333333333333:key/33333333-3333-3333-3333-333333333333'
+    To encrypt data, you must configure :class:`KMSMasterKeyProvider` with at least one CMK.
+    If you configure :class:`KMSMasterKeyProvider` with multiple CMKs,
+    it generates the data key using the first CMK and encrypts that data key using the rest,
+    so that the `encrypted message`_ includes a copy of the data key encrypted under each configured CMK.
+
+    .. _encrypted message: https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/message-format.html
+
+    >>> from aws_encryption_sdk.key_providers.kms import KMSMasterKeyProvider
+    >>> kms_key_provider = KMSMasterKeyProvider(key_ids=[
+    ...     "arn:aws:kms:us-east-1:2222222222222:key/22222222-2222-2222-2222-222222222222",
+    ...     "arn:aws:kms:us-east-1:3333333333333:key/33333333-3333-3333-3333-333333333333",
+    ... ])
+
+    You can also configure :class:`KMSMasterKeyProvider` with CMKs in multiple regions:
+
+    >>> from aws_encryption_sdk.key_providers.kms import KMSMasterKeyProvider
+    >>> kms_key_provider = KMSMasterKeyProvider(key_ids=[
+    ...     "arn:aws:kms:us-east-1:2222222222222:key/22222222-2222-2222-2222-222222222222",
+    ...     "arn:aws:kms:us-west-2:3333333333333:key/33333333-3333-3333-3333-333333333333",
+    ...     "arn:aws:kms:ap-northeast-1:4444444444444:key/44444444-4444-4444-4444-444444444444",
     ... ])
-    >>> kms_key_provider.add_master_key('arn:aws:kms:ap-northeast-1:4444444444444:alias/another-key')
 
-    .. note::
-        If no botocore_session is provided, the default botocore session will be used.
+    :class:`KMSMasterKeyProvider` needs AWS credentials in order to interact with `AWS KMS`_.
```

Comment:
Should we move the credentials related  information towards the top of the docstring (above line #85 (`To encrypt data, you must configure ...`) or is it okay either way? I am asking because conventionally information about how to set credentials appears before other information.

Reply:
I put the CMK configuration first because I expect more users to need to know the affect of CMK configuration than will need to know how to configure credentials. In the happy-path of "I'm running in AWS in an environment where the credentials magic in (EC2, ECS, Lambda, etc)", you usually don't need to think about the credentials at all; the default behavior will Just Work.

Reply:
Good point. Makes sense 👍